specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/Recommendations.json

```diff
@@ -0,0 +1,564 @@
+          }
+        },
+        "tags": [
+          "recommendations"
+        ],
+        "description": "Gets a list of all recommendations.",
+        "operationId": "GetRecommendations_List",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/RecommendationList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/recommendations/{recommendationId}": {
+      "get": {
+        "x-ms-examples": {
+          "Get a recommendation.": {
+            "$ref": "./examples/recommendations/GetRecommendation.json"
+          }
+        },
+        "tags": [
+          "recommendations"
+        ],
+        "description": "Gets a recommendation by its id.",
+        "operationId": "Get_SingleRecommendation",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/RecommendationId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Recommendation"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "patch": {
```

Comment:
>patch [](http://example.com/codeflow?start=7&length=5)

you cannot support only GET and PATCH, need to also have PUT\DELETE. 

If you need to only update selected properties of readonly resource then can you consider modelling it as PATCH instead ?

[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

Inconsistent or incomplete CRUD Any resource type MUST implement GET for collection and single resource regardless if it's a singleton or not. If your resource supports creation (PUT verb), it must support deletion (DELETE verb). Tracked resources must implement PATCH that allows updating tags at minimum. You may not have a PUT-only resource. While it is possible that your particular scenario does not require a particular CRUD operation, the platform demands it. There are core platform features such as Azure Policy, Azure Resource Graph, Template deployments, What-If and many others that rely on such API consistency. Do not break them.

Reply:
@raosuhas Those docs say "If your resource supports creation (PUT verb), it must support deletion (DELETE verb)" but we do not have a PUT operation. I don't see docs saying we need to support PUT and DELETE.  Had a review with the exact changes a while back, and this was not brought up -> https://github.com/Azure/azure-rest-api-specs-pr/pull/7705. We do have a patch operation though, could you clarify what you mean by "If you need to only update selected properties of readonly resource then can you consider modelling it as PATCH instead ?"